Common/lib/CANInterface/CANInterface.cpp

```diff
@@ -1,36 +1,44 @@
 #include "CANInterface.h"
 #include "Printing.h"
+#include "mbed.h"
+#include "rtos.h"
 
 #define CAN_PERIOD  10ms
 
-CANInterface::CANInterface(CAN &c, CANParser &cp, Ticker &tx, DigitalOut *stby, std::chrono::microseconds txPeriod) : can(c), canParser(cp), txTicker(tx)
-{
-    txTicker.attach(callback(this, &CANInterface::txHandler), (txPeriod==0s ? CAN_PERIOD : txPeriod));
-    can.attach(callback(this, &CANInterface::rxHandler), CAN::RxIrq);
-    if(stby)
-    {
+CANInterface::CANInterface(CAN &c, CANParser &cp, Thread &tx_thrd, Thread &rx_thrd, DigitalOut *stby, int txPeriod) : can(c), canParser(cp), tx_thread(rx_thrd), rx_thread(tx_thrd)
+{    
+    if(stby)    // should these be checked in the rx/tx threads? once we initialize this we 
+    {           // won't have control over it anymore 
         standby = stby;
         *standby = 0;   // active low
     }
 }
 
+void CANInterface::startCANTransmission(void)
+{
+    tx_thread.start(callback(this, &CANInterface::txHandler));
+    rx_thread.start(callback(this, &CANInterface::rxHandler));
+}
+
 // WARNING: This method will be called in an ISR context
-void CANInterface::rxHandler()
+void CANInterface::rxHandler(void)
 {
     CANMessage receivedCANMessage;
     while (can.read(receivedCANMessage))
     {
         canParser.parse(receivedCANMessage);
     }
+    ThisThread::sleep_for(CAN_PERIOD);
 }
 
 // WARNING: This method will be called in an ISR context
-void CANInterface::txHandler()
+void CANInterface::txHandler(void)
```

Comment:
Currently, this function will only run once to completion and then exit. Because this is a thread, we don't want that behavior (even though that was what we wanted when using `Ticker`). We don't want this function to ever return, so put the whole thing in an infinite while loop